src/dotnet/APIView/APIViewWeb/Services/GitHubClientFactory.cs

```diff
@@ -0,0 +1,137 @@
+
+    public async Task<GitHubClient> CreateGitHubClientAsync(string owner, string repository)
+    {
+        string appId = _configuration["GitHubApp:Id"];
+        string keyVaultUrl = _configuration["GitHubApp:KeyVaultUrl"];
+        string keyName = _configuration["GitHubApp:KeyName"];
+
+        if (string.IsNullOrEmpty(appId) || string.IsNullOrEmpty(keyVaultUrl) || string.IsNullOrEmpty(keyName))
+        {
+            _logger.LogWarning("GitHub App credentials not configured. GitHub client will not be available. " +
+                               "Required: GitHubApp:Id, GitHubApp:KeyVaultUrl, GitHubApp:KeyName");
+            return null;
+        }
+
+        try
+        {
+            string jwt = await CreateGitHubAppJwtAsync(keyVaultUrl, keyName, appId);
+            if (string.IsNullOrEmpty(jwt))
+            {
+                _logger.LogError("Failed to generate GitHub App JWT token.");
+                return null;
+            }
+
+            GitHubClient appClient = new(_productHeaderValue)
+            {
+                Credentials = new Credentials(jwt, AuthenticationType.Bearer)
+            };
+
+            Installation installation;
+
+            try
+            {
+                installation = await appClient.GitHubApps.GetRepositoryInstallationForCurrent(owner, repository);
+            }
+            catch (NotFoundException)
+            {
+                _logger.LogError("The GitHub App is not installed on the repository {Owner}/{Repository}.", owner,
+                    repository);
+                return null;
+            }
+
+            AccessToken accessToken = await appClient.GitHubApps.CreateInstallationToken(installation.Id);
+            GitHubClient installationClient = new(_productHeaderValue)
+            {
+                Credentials = new Credentials(accessToken.Token)
+            };
+
+            return installationClient;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Failed to create GitHub client for {Owner}/{Repository}", owner, repository);
+            return null;
+        }
+    }
+
+    private async Task<string> CreateGitHubAppJwtAsync(string keyVaultUrl, string keyName, string appId)
+    {
+        try
+        {
+            var header = new { alg = "RS256", typ = "JWT" };
+            long now = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+            var payload = new
+            {
+                iat = now,
+                exp = now + 600, // 10 minutes
+                iss = appId
+            };
+
+            string headerJson = JsonSerializer.Serialize(header);
+            string payloadJson = JsonSerializer.Serialize(payload);
+
+            string encodedHeader = Base64UrlEncode(headerJson);
+            string encodedPayload = Base64UrlEncode(payloadJson);
+            string unsignedToken = $"{encodedHeader}.{encodedPayload}";
+
+            KeyClient keyClient = new(new Uri(keyVaultUrl), new DefaultAzureCredential());
```

Comment:
We should avoid using DefaultAzureCredential and instead use an explicit Chained Credential that only contains the crendentials you expect to find for local development and for the live running service.